packages/amplify/amplify_flutter/lib/src/hybrid_impl.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:amplify_flutter/amplify_flutter.dart';
+
+/// {@template amplify_flutter.amplify_hybrid_impl}
+/// A hybrid implementation of [AmplifyClass] which uses method channels for
+/// iOS/Android plugins and Dart, otherwise.
+/// {@endtemplate}
+class AmplifyHybridImpl extends AmplifyClass {
+  /// {@macro amplify_flutter.amplify_hybrid_impl}
+  AmplifyHybridImpl() : super.protected();
+
+  final _methodChannelAmplify = MethodChannelAmplify();
+
+  @override
+  Future<void> configurePlatform(String config) async {
+    final amplifyConfig = AmplifyConfig.fromJson(
+      (jsonDecode(config) as Map<Object?, Object?>).cast(),
+    );
+    await Future.wait(
+      [
+        ...Auth.plugins,
+      ].map((p) => p.configure(config: amplifyConfig)),
+      eagerError: true,
+    );
+    if (!zIsWeb && (Platform.isAndroid || Platform.isIOS)) {
```

Comment:
*nit:* do you need this check since the whole class is only intended for use on iOS/Android?

Reply:
Nope!